src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/executor/SaveCheckerImpl.java

```diff
@@ -0,0 +1,165 @@
+    } catch (IOException e) {
+      throw new IllegalStateException(
+          String.format("Error reading path '%s'.", path.toString()), e);
+    }
+    ImmutableMap.Builder<String, ChunkCheckpoint> checkPointsMapBuilder = ImmutableMap.builder();
+    for (String scriptName : sortingColumnsMap.keySet()) {
+      if (fileMap.containsKey(scriptName) && !fileMap.get(scriptName).isEmpty()) {
+        List<Matcher> matchers = fileMap.get(scriptName);
+        // Note that the validator only validates information contained in the file names.
+        // The timestamp comparisons are only done lexicographically without checking whether each
+        // timestamp can be parsed. This is because the probability of having incorrectly generated
+        // timestamps in filenames while other aspects are normal is very low.
+        validateSortedChunkSequence(matchers);
+        Matcher lastMatcher = Iterables.getLast(matchers);
+        Instant lastInstant;
+        try {
+          lastInstant =
+              ZonedDateTime.of(
+                      chunkTimestampFormatter.parse(lastMatcher.group(3), LocalDateTime::from),
+                      ZoneOffset.UTC)
+                  .toInstant();
+        } catch (DateTimeParseException e) {
+          throw new IllegalStateException(
+              String.format(
+                  "Error parsing the last timestamp in the file name of the last chunk %s.",
+                  lastMatcher.group()),
+              e);
+        }
+        checkPointsMapBuilder.put(
+            scriptName,
+            ChunkCheckpoint.builder()
+                .setLastSavedInstant(lastInstant)
+                .setLastSavedChunkNumber(Integer.parseInt(lastMatcher.group(4)))
+                .build());
+      }
+    }
+    return checkPointsMapBuilder.build();
+  }
+
+  private static void validateSortedChunkSequence(List<Matcher> matchers) {
+    String prevEnding = "-10000000T000000S000000";
+    int prevChunkIndex = -1;
+    String prevFileName = "(beginning)";
+    String fileName;
+    for (Matcher matcher : matchers) {
+      fileName = matcher.group();
+      if (Integer.parseInt(matcher.group(4)) - prevChunkIndex != 1) {
+        throw new IllegalStateException(
+            String.format(
+                "The chunk index of file %s breaks the consecutiveness with other files (the"
+                    + " previous chunk number is %d), possibly indicating missing files."
+                    + " Aborting.",
+                fileName, prevChunkIndex));
+      }
+      if (matcher.group(2).compareTo(prevEnding) <= 0) {
+        throw new IllegalStateException(
+            String.format(
+                "The first time stamp of file %s is no later than the last time stamp of the"
+                    + " previous file %s, possibly indicating mixed runs. Aborting. ",
+                fileName, prevFileName));
+      }
+      if (matcher.group(3).compareTo(matcher.group(2)) < 0) {
```

Comment:
Making a function for each check seems a bit of overkill because these checks are only meaningful for two consecutive files from this particular sorted file list. But indeed To achieve the same goal of making it easier to scan the code, I added comments that turn them into meaningful visual blocks.